lms/static/js/dashboard/views/course_view.js

```diff
@@ -0,0 +1,147 @@
+;(function (define) {
+    'use strict';
+
+    define(['jquery', 'backbone', 'underscore', 'logger'],
+        function ($, Backbone, _, Logger) {
+            var CourseView = Backbone.View.extend({
+
+                events: {
+                    'click .action-more': 'toggleCourseActionsDropdown',
+                    'click .action-unenroll': 'unEnroll',
+                    'click .action-email-settings': 'emailSettings',
+                    'click #upgrade-to-verified': 'upgradeToVerified',
+                    'click #block-course-msg a[rel="leanModal"]': 'unRegisterBlockCourse'
+                },
+
+                initialize: function (options) {
+                    this.setElement(this.el);
+                    this.listenTo(options.tabbedView, 'rendered', this.rendered);
+                },
+
+                rendered: function () {
+                    var $actionUnroll = this.$('.action-unenroll');
+                    var $unRegisterBlockCourse = this.$('#unregister_block_course');
+
+                    this.bindUnEnrollModal($actionUnroll);
+                    this.bindUnEnrollModal($unRegisterBlockCourse);
+                    this.bindEmailSettingsModal();
+                },
+
+                bindUnEnrollModal: function ($selector) {
+                    $selector.leanModal({
+                        overlay: 1,
+                        closeButton: ".close-modal"
+                    });
+
+                    var id = _.uniqueId('unenroll-');
+                    $selector.attr('id', id);
+                    var trigger = "#" + id;
+
+                    accessible_modal(
+                        trigger,
+                        "#unenroll-modal .close-modal",
+                        "#unenroll-modal",
+                        "#dashboard-main"
+                    );
+                },
+
+                bindEmailSettingsModal: function () {
+                    var $actionEmailSettings = this.$('.action-email-settings');
+
+                    $actionEmailSettings.leanModal({
+                        overlay: 1,
+                        closeButton: ".close-modal"
+                    });
+
+                    var id = _.uniqueId('email-settings-');
+                    $actionEmailSettings.attr('id', id);
+                    var trigger = "#" + id;
+
+                    accessible_modal(
+                        trigger,
+                        "#email-settings-modal .close-modal",
+                        "#email-settings-modal",
+                        "#dashboard-main"
+                    );
+                },
+
+                toggleCourseActionsDropdown: function (e) {
+                    var index = this.$(e.currentTarget).data('dashboard-index');
+
+                    // Toggle the visibility control for the selected element and set the focus
+                    var $dropDown = this.$('div#actions-dropdown-' + index);
+
+                    $dropDown.toggleClass('is-visible');
```

Comment:
It seems weird that you are toggle the class and then have an if statement based off said toggle. It would make more sense to toggle after the if statement incase there is a race condition where the class has not been toggled in time to be catch in the subsequent if statement (i.e. test state THEN change state).